includes/amp-helper-functions.php

```diff
@@ -301,6 +301,56 @@ function amp_add_frontend_actions() {
 	add_action( 'wp_head', 'amp_add_amphtml_link' );
 }
 
+/**
+ * Determine whether AMP is available for the current URL.
+ *
+ * @todo This should go into the AmpProject\AmpWP namespace?
+ * @since 1.6
+ *
+ * @return bool True if there is an AMP version, false otherwise.
+ */
+function is_amp_available() {
+	global $wp_query;
+
+	if ( ! $wp_query instanceof WP_Query || ! did_action( 'wp' ) ) {
+		_doing_it_wrong(
+			__FUNCTION__,
+			esc_html(
+				sprintf(
+						/* translators: %1$s: function name, %2$s: the current action, %3$s: the wp action, %4$s: the WP_Query class, %5$s: the amp_skip_post() function */
+					__( '%1$s was called too early and so it will not work properly. WordPress is currently doing the "%2$s" action. Calling this function before the "%3$s" action means it will not have access to %4$s and the queried object to determine if it is an AMP response, thus neither the "%5$s" filter nor the AMP enabled toggle will be considered.', 'amp' ),
+					__FUNCTION__ . '()',
+					current_action(),
+					'wp',
+					'WP_Query',
+					'amp_skip_post()'
+				)
+			),
+			'1.6.0'
+		);
+		return false;
+	}
+
+	$queried_object = get_queried_object();
+	if ( current_theme_supports( AMP_Theme_Support::SLUG ) ) {
+		// Abort if in Transitional mode and AMP is not available for the URL.
+		$availability = AMP_Theme_Support::get_template_availability( $wp_query );
+		return $availability['supported'];
```

Comment:
This isn't actually the final word… 

Consider the case in Transitional mode where there is invalid markup that is kept. In that case, AMP should actually be blocked entirely, as seen in `amp_add_amphtml_link()`:

https://github.com/ampproject/amp-wp/blob/722f8c9612b2e831217ed9e72ba50caa421f3553/includes/amp-helper-functions.php#L531-L552

Reply:
In the same way, when someone does end up on a paired AMP page that has kept invalid markup, a redirect is performed to the non-AMP version:

https://github.com/ampproject/amp-wp/blob/722f8c9612b2e831217ed9e72ba50caa421f3553/includes/class-amp-theme-support.php#L2114-L2125

This URL will need to have the `noamp` query param added to prevent an infinite loop.

This would be addressing this todo here: https://gist.github.com/westonruter/5f4f27ed3bdd676e3f08524b9bab8bb4#file-amp-mobile-js-redirection-php-L103

Reply:
Addressed in https://github.com/ampproject/amp-wp/pull/4796/commits/85ab507c2149ebdb93f5e4ee8b48472888128c2b.